pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/IngestionDelayTracker.java

```diff
@@ -351,6 +398,28 @@ public void timeoutInactivePartitions() {
     }
   }
 
+  /**
+   * Updates the latest offsets for each partition at a configurable frequency to reduce load.
+   */
+  private void updateLatestOffsets() {
+    if (!_isServerReadyToServeQueries.get() || _realTimeTableDataManager.isShutDown()) {
+      return;
+    }
+    for (Map.Entry<Integer, IngestionInfo> entry : _ingestionInfoMap.entrySet()) {
+      int partitionId = entry.getKey();
+      IngestionInfo ingestionInfo = entry.getValue();
+      Supplier<StreamPartitionMsgOffset> latestOffsetFetcher = ingestionInfo._latestOffsetFetcher;
+      if (latestOffsetFetcher != null) {
```

Comment:
latestOffsetFetcher cannot be null right? What can be null is the value returned by it

Reply:
Actually it is null in some tests, hence the check